### PR TITLE
Doc | Add CONTRIBUTING.md + extending leverage cli section

### LINKED ISSUE #134: Enhancement | Add CONTRIBUTING.md
## Describe the Enhancement

Add a contribution guideline file (CONTRIBUTION.md) to this repo.

## Expected Behavior

Any user trying to contribute to this project has to be a clear contribution guideline at hand.

## Use Case

Since this repository can be contributed by the community, there's a need of having a clear contribution guideline.

## Describe Ideal Solution

Add the file following well known and tested guidelines as specified in the References.

## Alternatives Considered

n/a

### References

* [Feature | Open Source repository  le-tf-infra-aws#223](https://github.com/binbashar/le-tf-infra-aws/issues/223)
* Probably linking to and standard doc (https://leverage.binbash.com.ar/work-with-us/contribute/)
* https://github.com/binbashar/le-tf-infra-aws/community
* https://github.com/binbashar/le-tf-infra-aws/blob/master/CONTRIBUTING.md

### LINKED ISSUE #143: Doc |  Add "Extending Leverage" section
### Description

A new section called **_"Extending Leverage"_** that includes information about the 4 mechanisms available for extending and configuring [Leverage Reference Architecture environments](https://github.com/binbashar/le-tf-infra-aws). This section should summarize the following topics:

1. Using `build.env` to override defaults such as the Terraform image that is used by the CLI
2. Using additional `.tfvars` configuration files at the account level or at the global level to keep your configuration entries organized
3. Using a `build.py` to define your own tasks via Python
4. Forking https://github.com/binbashar/leverage and start contributing
 
#### Why?
- This feature would greatly improve the user experience by making it easy for users to find and understand the information they need to customize, extend and configure their Leverage cli orchestrated Ref Arch environments. 
- It would also help to ensure that users are able to take full advantage of the capabilities offered by Leverage.
- Avoid having the information spread out in different parts of the documentation
- Having a dedicated section for this topic

### Expected Behavior
The new section should be easy to find and understand for users who are looking for information about how to extend and configure their Leverage cli orchestrated environments. It should include clear explanations and examples for each of the 4 mechanisms mentioned above.

### Use Case
A user wants to customize their Leverage environment by overriding defaults, using additional configuration files, and creating their own tasks. They are looking for clear and concise information about how to do this.

### Ideal Solution
The ideal solution would be a new section called "Extending Leverage" that includes information about the 4 mechanisms available for extending and configuring Leverage environments. This section should be easy to find and understand for users who are looking for information about how to extend and configure their Leverage environments. It should include clear explanations and examples for each of the 4 mechanisms mentioned above.

Main section under extension summarizing the 4 mechanisms available to extend and confiture leverage environments

- ✅ Using `build.env` to override defaults such as the Terraform image that is used by the CLI
  - Mention build.env here => [https://leverage.binbash.com.ar/user-guide/ref-architecture-aws/configs/](https://leverage.binbash.com.ar/user-guide/ref-architecture-aws/configs/ "‌")
  - Add more details at user-guide/leverage-cli/exgending-leverage
    - Entry with `build.env` info ([https://github.com/binbashar/leverage/blob/master/leverage/conf.py](https://github.com/binbashar/leverage/blob/master/leverage/conf.py "‌"))
    - Move this existent section under extending-leverage too [https://leverage.binbash.com.ar/user-guide/leverage-cli/tasks/](https://leverage.binbash.com.ar/user-guide/leverage-cli/tasks/ "‌")
- ✅ Using additional `.tfvars` configuration files at the account level or at the global level to keep your configuration entries organized
  - [https://leverage.binbash.com.ar/user-guide/ref-architecture-aws/dir-structure/](https://leverage.binbash.com.ar/user-guide/ref-architecture-aws/dir-structure/ "‌")
    - Add context about config files and its purpose
  - [https://leverage.binbash.com.ar/user-guide/ref-architecture-aws/configs/](https://leverage.binbash.com.ar/user-guide/ref-architecture-aws/configs/ "‌")
    - Add [common-variables.tf](http://common-variables.tf "‌") and complementary info for the project structure section ([https://github.com/binbashar/le-tf-infra-aws/blob/master/config/common-variables.tf](https://github.com/binbashar/le-tf-infra-aws/blob/master/config/common-variables.tf "‌"))
- ✅ Using a [build.py](http://build.py "‌") to define your own tasks via Python
  \- add under extend-leverage setion
  \- [https://leverage.binbash.com.ar/user-guide/leverage-cli/tasks/](https://leverage.binbash.com.ar/user-guide/leverage-cli/tasks/ "‌")
- ✅ Fork [github.com/binbashar/leverage](http://github.com/binbashar/leverage "‌") and start contributing
  \- Contribute MD per repo Ref Arch + CLI
  \- Consolidated Contribute seciton
  \- [https://leverage.binbash.com.ar/work-with-us/contribute/](https://leverage.binbash.com.ar/work-with-us/contribute/ "‌")
  
  ---
  CC: @diego-ojeda-binbash @juanmatias @angelofenoglio

### PR DESCRIPTION
**Pull request by exequielrafaela**
## What?

- [Add CONTRIBUTING.md](https://github.com/binbashar/le-ref-architecture-doc/commit/e1335629bb1b4f1e0c3f60984e72e9ae4f9e928b) - @[exequielrafaela](https://github.com/binbashar/le-ref-architecture-doc/commits?author=exequielrafaela)
- [removing old diagrams from how it works organization accounts section](https://github.com/binbashar/le-ref-architecture-doc/commit/07351e893af8aba5fc3dbf2beed0ce6fcb71ead3) - @[exequielrafaela](https://github.com/binbashar/le-ref-architecture-doc/commits?author=exequielrafaela)
- [minor sections updates](https://github.com/binbashar/le-ref-architecture-doc/commit/8c8a71ecc5ffe5d9c313f466b6d69e8936a6df3c) - @[exequielrafaela](https://github.com/binbashar/le-ref-architecture-doc/commits?author=exequielrafaela)
- [Adding leverage-cli/extending-leverage section](https://github.com/binbashar/le-ref-architecture-doc/commit/0356c3acfc88f433fe3186fb2f53ab7cb7b18fa9) - @[exequielrafaela](https://github.com/binbashar/le-ref-architecture-doc/commits?author=exequielrafaela)

## Why?
- closes #134 
- closes #143

### CODE REVIEW

File: docs/user-guide/leverage-cli/extending-leverage/extending.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Extending & Configuring leverage CLI
+
+## Override defaults via `build.env` file
+By utilizing the [`build.env` capability](https://github.com/binbashar/leverage/blob/master/leverage/conf.py), you can
+easily change some default behaviors of the CLI. This allows you to specify several configurations for the CLI, 
+like the
```
Comment (juanmatias) [inline docs/user-guide/leverage-cli/extending-leverage/extending.md:6]:
```suggestion
such as the
```

File: docs/user-guide/leverage-cli/extending-leverage/extending.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+configuration entries well-organized. You can have separate files for different accounts or environments, making it easy
+to manage and maintain your infrastructure. This also makes it easier for other team members to understand and work with
+your configuration, reducing the risk of misconfigurations or errors.
+
+!!! info "Read More about `.tfvars` config files"
+    In order to further understand this mechanism and how to use it please visit the dedicated
+    [.tfvars configs](../../ref-architecture-aws/configs.md) entry.
+
+## Custom tasks with build.py 
+Leverage CLI has a native mechanism to allow yo customizing your workflow. With the custom tasks feature using `build.py`,
```
Comment (juanmatias) [inline docs/user-guide/leverage-cli/extending-leverage/extending.md:28]:
```suggestion
Leverage CLI has a native mechanism to allow customizing your workflow. With the custom tasks feature using `build.py`,
```

**Review by juanmatias — COMMENTED** [review]
a few comments

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+When contributing to this repository, please first discuss the change you wish to make via issue
+with the owners of this repository before making a change.
+
+Please note we have a code of conduct, please follow it in all your interactions with the project.
+
+## Pull Request Process
+
+1. Ensure any install or build dependencies are removed before the end of the layer when doing a
+   build.
```
[the remark this answers is not in the document]
Comment (exequielrafaela) [reply CONTRIBUTING.md:11]: Thanks!

**Comment by exequielrafaela** [comment]
> a few comments

@juanmatias thanks for your review and all your commit suggestions. Much appreciated!!!

**Pull request merged**